Спринт 9/25

boris-sulyagin opened the pull request.

boris2034 committed:
Спринт 9/25 → Тема 1/5: Стили и темы → Урок 3/3

boris2034 committed:
Спринт 9/25 → Тема 2/5: Context и Intent → Урок 3/3

boris2034 committed:
Спринт 9/25 → Тема 3/5: EditText и TextWatcher → Урок 3/3

boris2034 committed:
Спринт 9/25 → Тема 4/5: Хранение данных. Начало → Урок 4/4

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -1,23 +1,65 @@
+    override fun onSaveInstanceState(outState: Bundle) {
+        // put string value
+        super.onSaveInstanceState(outState)
+        outState.putString("string_value", string)
```
Ключи для передачи данных лучше вынести в константу, т.к. они используются в более чем 1 месте. Ещё один плюс вынесения в константы - в будущем легче будет этот экран переиспользовать

ArturNurtdinov reviewed (commented):
1. Заголовок главного экрана попадает под системный статус бар и отступ между кнопкой и заголовком должен быть 36dp
![image](https://github.com/user-attachments/assets/98520844-d6d0-44a5-94a2-9f6a97acd03e)
Макет:
![image](https://github.com/user-attachments/assets/5270de87-6612-4a6b-a978-3a19a599b967)
2. Системный статус бар на экране настроек всегда белого цвета, из-за чего не видно его контента и отступ между заголовком "Настройки" и "Тёмная тема" слишком большой:
![image](https://github.com/user-attachments/assets/a494239e-c71d-4e47-9964-b0aff024b2a0)
Макет:
![image](https://github.com/user-attachments/assets/c5b4851c-925f-42cf-a26e-a9d89a264924)
3. На экране настроек заголовок залезает под систменый статус бар и поле ввода не соответствует макетам (высота и цвет текста):
![image](https://github.com/user-attachments/assets/63a01b6d-e955-4120-905b-396cd5e48cc0)
Макет:
![image](https://github.com/user-attachments/assets/92e3aa92-99d6-4abb-a66c-9f36775247ab)

boris2034 committed:
Спринт 9/25 Правки по комментариям.

- В файле вёрстки экрана EditText и кнопка «Очистить» находятся в одном FrameLayout (кнопка поверх поля ввода) - Используется LinearLayout
- Цвета, отступы и шрифты соответствуют дизайну (скриншоты на гитхабе):
- Заголовок главного экрана попадает под системный статус бар и отступ между кнопкой и заголовком должен быть 36dp
- Системный статус бар на экране настроек всегда белого цвета, из-за чего не видно его контента и отступ между заголовком "Настройки" и "Тёмная тема" слишком большой
- На экране настроек заголовок залезает под систменый статус бар и поле ввода не соответствует макетам (высота и цвет текста)
- Цвет каретки в поле ввода не соответствует макетам. Должен быть синий
- Нажатие на кнопку «Очистить» удаляет текст в поле ввода поискового запроса и скрывает клавиатуру - Клавиатура не скрывается
- Нажатие кнопки «Назад» возвращает пользователя на предыдущий экран - На экране настроек кнопка не нажимается
- Переопределён метод onRestoreInstantState() для помещения сохранённого текста в переменную для хранения текста поискового запроса. Восстановленный текст помещается в EditText - Метод закомментирован

boris2034 committed:
Спринт 9/25 Правки по комментариям.

- Переопределён метод onRestoreInstantState() для помещения сохранённого текста в переменную для хранения текста поискового запроса. Восстановленный текст помещается в EditText - Метод закомментирован

boris2034 committed:
Вернул приватность переменной
